src/components/Tabset/index.js

```diff
@@ -76,6 +83,22 @@ export default class Tabset extends Component {
             this.updateButtonsVisibility();
             this.isFirstTime = false;
         }
+
+        if (variant === 'linear') {
+            const tabIndex = getTabIndexFromName(tabsetChildren, activeTabName);
+            if (tabIndex !== -1) {
+                const tabset = this.tabsetRef.current;
+                const currentTabsetMeta = getTabsMeta(tabset);
+                const currentTabMeta = getTabMeta(activeTabName, tabsetChildren);
+                if (
+                    activeTabName !== prevProp.activeTabName ||
+                    tabMeta.left !== currentTabMeta.left ||
+                    tabsetMeta.width !== currentTabsetMeta.width
+                ) {
+                    this.updateIndicator();
+                }
+            }
+        }
```

Comment:
I think it is more simply pass the logic of painting the new variant to the Tab component, this avoiding doing all this computation.
If it is complex to mix the two variants then you can do in Tab/index.js:
```
if (variant === 'line') {
    return ...
}
return ...
```
then you can render two different components based on the variant